src/domains/user/queries/useAddPreferenceMutation.tsx

```diff
@@ -0,0 +1,32 @@
+import { useRouter } from 'next/navigation';
+import { useMutation } from '@tanstack/react-query';
+import { PreferenceType } from '@/domains/user/types/profile';
+import PATH from '@/shared/constants/path';
+import userService from '@/domains/user/queries/service';
+import useToast from '@/shared/hooks/useToast';
+import ToastPop from '@/shared/components/ToastPop';
+
+const useAddPreferenceMutation = () => {
+  const { openToast } = useToast();
+  const { push } = useRouter();
+
+  const mutationFn = async (preference: Array<PreferenceType>) => {
+    await userService.addPreference(preference);
+  };
```

Comment:
```suggestion
  const mutationFn = userService.addPreference
```
더 짧게도 쓸 수 있어요!
~~사실 저도 이렇게 쓴 부분 많음~~

Reply:
저도 처음에 저렇게 작성을 했었는데 무슨 undefined 하다면서 에러를 던지더라구요.
`const mutationFn = userService.addPreference` 이렇게 작성하면 preference 값이 안 넘어가는 것 같아서 저렇게 작성해줬습니다..!

Reply:
엇 그런가요?